src/main/java/org/primeframework/mvc/content/json/BaseJacksonContentHandler.java

```diff
@@ -211,12 +211,17 @@ private void addFieldError(InvalidFormatException e) {
       String messageText = e.getMessage();
       Matcher matchesEnumNotValidValue = invalidEnumerationValue.matcher(messageText);
       String message = null;
-      if (matchesEnumNotValidValue.matches()) {
+      if (matchesEnumNotValidValue.matches() && matchesEnumNotValidValue.groupCount() == 2) {
         code = "[invalid]%s".formatted(field);
         // if we have an invalid enum value, provide a better message
+        String possibleValues = matchesEnumNotValidValue.group(2);
+        String valueUsed = matchesEnumNotValidValue.group(1);
```

Comment:
🟩 Don't we always expect that it has quotes around the value? Is it valid if it doesn't have quotes around the value? Could the regex be updated to remove the quotes from the value grouping instead of filtering them here?